griptape/drivers/prompt/anthropic_prompt_driver.py

```diff
@@ -87,44 +99,121 @@ def _base_params(self, prompt_stack: PromptStack) -> dict:
             return message.to_text()
         else:
-            return [self.__prompt_stack_content_message_content(content) for content in message.content]
+            return [self.__to_anthropic_message_content(content) for content in message.content]
 
-    def __prompt_stack_content_message_content(self, content: BaseMessageContent) -> dict:
+    def __to_anthropic_message_content(self, content: BaseMessageContent) -> dict:
         if isinstance(content, TextMessageContent):
-            return {"type": "text", "text": content.artifact.to_text()}
+            return {"type": "text", "text": content.artifact.value}
         elif isinstance(content, ImageMessageContent):
+            artifact = content.artifact
+
             return {
                 "type": "image",
-                "source": {"type": "base64", "media_type": content.artifact.mime_type, "data": content.artifact.base64},
+                "source": {"type": "base64", "media_type": artifact.mime_type, "data": artifact.base64},
+            }
+        elif isinstance(content, ActionCallMessageContent):
+            action = content.artifact.value
+
+            return {"type": "tool_use", "id": action.tag, "name": f"{action.name}_{action.path}", "input": action.input}
+        elif isinstance(content, ActionResultMessageContent):
+            artifact = content.artifact
+
+            if isinstance(artifact, ListArtifact):
+                message_content = [self.__to_anthropic_tool_result_content(artifact) for artifact in artifact.value]
+            else:
+                message_content = [self.__to_anthropic_tool_result_content(artifact)]
+
+            return {
+                "type": "tool_result",
+                "tool_use_id": content.action.tag,
+                "content": message_content,
+                "is_error": isinstance(artifact, ErrorArtifact),
             }
         else:
             raise ValueError(f"Unsupported prompt content type: {type(content)}")
 
-    def __message_content_to_prompt_stack_content(self, content: ContentBlock) -> BaseMessageContent:
+    def __to_anthropic_tool_result_content(self, artifact: BaseArtifact) -> dict:
+        if isinstance(artifact, ImageArtifact):
+            return {
+                "type": "image",
+                "source": {"type": "base64", "media_type": artifact.mime_type, "data": artifact.base64},
+            }
+        elif isinstance(artifact, (TextArtifact, ErrorArtifact, InfoArtifact)):
+            return {"type": "text", "text": artifact.to_text()}
+        else:
+            raise ValueError(f"Unsupported tool result artifact type: {type(artifact)}")
+
+    def __to_prompt_stack_message_content(self, content: ContentBlock) -> BaseMessageContent:
         if content.type == "text":
             return TextMessageContent(TextArtifact(content.text))
+        elif content.type == "tool_use":
+            name, path = content.name.split("_", 1)
```

Comment:
Is it possible for `action.name` to contain an underscore?

Reply:
Unfortunately, yes. But the LLM providers are very particular about what characters can be included in the function name, and none of them support nested routes like we do (name/path).

The only thing protecting us here is that Actions are always Tools (right now at least), which are classes that _should_ follow python's class [naming convention](https://peps.python.org/pep-0008/#class-names). But this is a weak protection.

@vasinov we said we'd figure this out on the PR, so let's use this thread.

Reply:
I think that is a reasonable convention, so long as we call out somewhere in the Action/Tool docs that class names must not contain an underscore in order to properly work with native functions. Actually, you could ensure correctness by raising an exception if `action.name` contains an underscore.

That said, there are a couple places where split is called without the maxsplits arg. We should update those so that we don't need the same restriction on `path`.

Reply:
I replaced all instances of `split` with a utility so there should be more consistency.

Will update the docs to call out the naming.